opennms-config/src/test/java/org/opennms/netmgt/config/SnmpPeerFactoryTest.java

```diff
@@ -28,33 +28,57 @@
 
 package org.opennms.netmgt.config;
 
-import static org.junit.Assert.assertThat;
-
+import java.io.IOException;
 import java.net.InetAddress;
 import java.net.UnknownHostException;
 import java.util.List;
 
 import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
 import org.opennms.core.test.MockLogAppender;
+import org.opennms.core.test.OpenNMSJUnit4ClassRunner;
 import org.opennms.core.utils.InetAddressUtils;
 import org.opennms.core.utils.LocationUtils;
+import org.opennms.features.config.dao.impl.util.JaxbXmlConverter;
 import org.opennms.netmgt.config.snmp.SnmpProfile;
 import org.opennms.netmgt.snmp.SnmpAgentConfig;
-import org.springframework.core.io.ByteArrayResource;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.test.context.ContextConfiguration;
+
+import static org.junit.Assert.*;
 
-import junit.framework.TestCase;
+@RunWith(OpenNMSJUnit4ClassRunner.class)
+@ContextConfiguration(locations={
+        "classpath:etc/mock-dao.xml"
+})
 
-public class SnmpPeerFactoryTest extends TestCase {
+public class SnmpPeerFactoryTest  {
 
     private int m_version;
 
-    @Override
-    protected void setUp() throws Exception {
+    @Before
+    public void setUp() throws Exception {
         setVersion(SnmpAgentConfig.VERSION2C);
-        SnmpPeerFactory.setInstance(new SnmpPeerFactory(new ByteArrayResource(getSnmpConfig().getBytes())));
+        this.updateConfig(getSnmpConfig());
         MockLogAppender.setupLogging(true);
     }
 
+    @Autowired
+    SnmpPeerFactory snmpPeerFactory;
+
+    private void updateConfig(String configXml) throws IOException {
+        try {
+            JaxbXmlConverter converter = new JaxbXmlConverter("snmp-config.xsd", "snmp-config", null);
+            String json = converter.xmlToJson(configXml);
+            snmpPeerFactory.updateConfig(json);
+            snmpPeerFactory.reload();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
why not `throws Exception` instead of catching it?

Reply:
removed